common/views/components/LoadingIndicator/LoadingIndicator.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import { useEffect } from 'react';
+import NProgress from 'nprogress';
+import Router from 'next/router';
+import theme from '../../themes/default';
+
+const LoadingIndicator = () => {
+  function handleRouteChangeStart() {
+    NProgress.start();
+  }
+
+  function handleRouteChangeComplete() {
+    NProgress.done();
+  }
+  useEffect(() => {
+    Router.events.on('routeChangeStart', handleRouteChangeStart);
+    Router.events.on('routeChangeComplete', handleRouteChangeComplete);
+
+    return () => {
+      Router.events.off('routeChangeStart', handleRouteChangeStart);
+      Router.events.off('routeChangeComplete', handleRouteChangeComplete);
+    };
+  }, []);
+  const color = theme.colors.yellow;
+  const height = 2;
+
+  NProgress.configure({ showSpinner: false });
+
+  return (
```

Comment:
Hmmmm. Mixing styledJSX and styled-components doesn't feel like it's inline with all the great work you've done simplifying things?

Reply:
I was wondering about this, yeah – but wasn't sure what a reasonable solution was in light of the fact that the component doesn't render any markup itself.

Reply:
https://github.com/sergiodxa/next-nprogress
❓ Could just nab how they've done it too if we want more control.

Reply:
I'm not sure that gets us any more control, particularly – just a different way of doing it.

I opted for rendering an element from the `LoadingIndicator` component, and using that as the `parent` configuration option to `NProgress`.